vm/src/obj/objfloat.rs

```diff
@@ -575,7 +575,30 @@ impl PyFloat {
             "nan" => Ok(std::f64::NAN),
             "inf" => Ok(std::f64::INFINITY),
             "-inf" => Ok(std::f64::NEG_INFINITY),
-            _ => Err(vm.new_value_error("invalid hexadecimal floating-point string".to_string())),
+            value => {
+                let err_msg = "invalid hexadecimal floating-point string";
+                let res = if value.contains("0x") {
+                    let mut hex = "".to_string();
+                    for ch in value.chars() {
+                        if ch == 'p' {
+                            hex += ".p";
+                        } else {
+                            hex += &ch.to_string();
+                        }
+                    }
+
+                    match hexf_parse::parse_hexf64(&hex, false) {
+                        Ok(f) => Ok(f),
+                        _e => Err(vm.new_value_error(err_msg.to_string())),
+                    }
+                } else {
+                    match value.parse::<f64>() {
+                        Ok(f) => Ok(f),
```

Comment:
You could maybe use the `map_err` function here.

Reply:
Thanks for your advice!
I changed the whole code on match statement to simplyfied one.

- Use `or_else` instead of value condition and `match` statement
  - For floating-point input
- Use `map_err` instead of `match ` statement
  - For hexadecimal floating-point input without dot